cmake/cuda.cmake

```diff
@@ -21,8 +21,21 @@
 macro(find_cuda use_cuda)
   set(__use_cuda ${use_cuda})
   if(__use_cuda STREQUAL "ON")
-    cmake_policy(SET CMP0146 OLD)
-    find_package(CUDA QUIET)
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.17")
+      find_package(CUDAToolkit QUIET)
+      if(CUDAToolkit_FOUND)
+        set(CUDA_FOUND TRUE)
+        set(CUDA_INCLUDE_DIRS ${CUDAToolkit_INCLUDE_DIRS})
+        set(CUDA_TOOLKIT_ROOT_DIR ${CUDAToolkit_ROOT})
+        set(CUDA_CUDA_LIBRARY ${CUDAToolkit_LIBRARY_DIR}/libcuda.so)
+        set(CUDA_CUDART_LIBRARY ${CUDAToolkit_LIBRARY_DIR}/libcudart.so)
+        set(CUDA_NVRTC_LIBRARY ${CUDAToolkit_LIBRARY_DIR}/libnvrtc.so)
+        set(CUDA_CUDNN_LIBRARY ${CUDAToolkit_LIBRARY_DIR}/libcudnn.so)
```

Comment:
Let's remove the cudnn and cublas lines---we have removed the cudnn/cublas dependencies in mpk.